core/src/main/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilder.java

```diff
@@ -96,35 +147,73 @@ public GeoBoundingBoxQueryBuilder bottomRight(String geohash) {
      * @param lon The longitude
      */
     public GeoBoundingBoxQueryBuilder bottomLeft(double lat, double lon) {
-        box[BOTTOM] = lat;
-        box[LEFT] = lon;
+        if (topLeft != null) {
+            topLeft.resetLon(lon);
+        } else {
+            topLeft = new GeoPoint(Double.NaN, lon);
+        }
+
+        if (bottomRight != null) {
+            bottomRight.resetLon(lon);
```

Comment:
I might be confused, but shouldn't this set the lat? At least when bottomRight is not here yet you create a point where lat is set, right?